driver/controller.go

```diff
@@ -94,30 +92,57 @@ func newControllerService(client *v3.Client, nodeMeta *nodeMetadata) controllerS
 // This function is idempotent.
 func (d *controllerService) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest) (*csi.CreateVolumeResponse, error) {
 	klog.V(4).Infof("CreateVolume")
-	// TODO(multizone cluster) use req.AccessibilityRequirements,
-	// To create block storage volume in the right zone.
 
-	// TODO(multizone cluster) fetch all zone
-	volumes, err := d.client.ListBlockStorageVolumes(ctx)
+	zones, err := d.client.ListZones(ctx)
 	if err != nil {
-		klog.Errorf("create block storage volume list: %v", err)
+		klog.Errorf("create block storage volume list zones: %v", err)
 		return nil, err
 	}
 
-	// Make the call idempotent since CreateBlockStorageVolume is not.
-	for _, v := range volumes.BlockStorageVolumes {
-		if v.Name == req.Name {
-			return &csi.CreateVolumeResponse{
-				Volume: &csi.Volume{
-					VolumeId: exoscaleID(d.zoneName, v.ID),
-					// API reply in bytes then send it without conversion
-					CapacityBytes:      v.Size,
-					AccessibleTopology: newZoneTopology(d.zoneName),
-				},
-			}, nil
+	// Fetch all zones volumes
+	for _, zone := range zones.Zones {
+		client := d.client.WithEndpoint(zone.APIEndpoint)
+
+		resp, err := client.ListBlockStorageVolumes(ctx)
+		if err != nil {
+			// TODO: remove it when Block Storage is available in all zone.
+			if strings.Contains(err.Error(), "Availability of the block storage volumes") {
+				continue
+			}
+
+			klog.Errorf("create block storage volume list: %v", err)
+			return nil, err
+		}
+
+		// Make the call idempotent since CreateBlockStorageVolume is not.
+		for _, v := range resp.BlockStorageVolumes {
+			if v.Name == req.Name {
+				return &csi.CreateVolumeResponse{
+					Volume: &csi.Volume{
+						VolumeId: exoscaleID(zone.Name, v.ID),
+						// API reply in bytes then send it without conversion
+						CapacityBytes:      v.Size,
+						AccessibleTopology: newZoneTopology(zone.Name),
+					},
+				}, nil
+			}
```

Comment:
I think this change will now prevent users from creating a volume with the same name in different zones. But is this a real constraint of block storage? Do block storage volume names have to be unique across zones?
If no, then we should make this idempotent for the zone of the cluster only.

Reply:
Maybe this question can be gone if we decide to do this:
https://github.com/exoscale/exoscale-csi-driver/pull/9/files/908ce54b376e423f416cabd33633ee64fd19e6cd#r1481087895